README.md

```diff
@@ -22,7 +18,73 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+The gem basically handles the conversion through the method `parse` defined in the `JsonToGraphql` module that gets passed a json in the following format.
+
+```ruby
+    {
+      "query" => {
+        "_variables" => {
+          "x" => 1, "y" => 2
+        },
+        "libraries" => {
+          "_args" => {
+            "id" => 123
+          }, 
+          "_attrs" => [:name,:address], 
+          "books" => {
+            "_args" => {
+              "price" => 40
+            }, 
+            "_attrs" => [:price,:name, :author]
+          }, 
+          "employees" => {
+            "_args" => {
+                "department" => "history"
+            },
+            "_attrs" => [:name, :age, :department]
+          }
+        }
+      }
+    }
+```
+
+Assuming this is stored in a variable called `json_query` in `JSON` format then
+
+```ruby
+JsonToGraphql.parse(json_query)
+```
+should output
+
+```graphql
+  {
+    query($x: 1,$y: 2) {
+      libraries(id: 123) {
+        name
+        address
+        devices(price: 40) {
+          price
+          name
+          author
+        }
+        employees(department: history){
+          name
+          age
+          department
+        }
+      }
+    }
+  }
+```
+
+Here each key/node can have one of 4 options:
+
+1 - **Variables**: Using the key `_variables` which are local variables that you want to use later in the query, usually placed on the top key defining the entire query, in this case this key is `"query"`.
+
+2 - **Arguments**: Using the key `_args` which is another hash containing arguments passed to a node like in `devices(price: 40)`.
```

Comment:
Thoughts on 

"Use the `_args` key to define the arguments to be passed to your GraphQL engine when fetching the relation."